### PR TITLE
Cleaning up stl includes. (Starting With Universal_System)

### PR DESCRIPTION
**Pull request by fundies**
Still including:
deque GAME_GLOBALS.H
cmath mathnc.h
math.h scalar.h
stdint.h, vector fonts_internal.h
vector, map timline_object.h
stdint.h, vector sprites_internal.h

### CODE REVIEW

File: ENIGMAsystem/SHELL/Universal_System/darray.h [hunk 1]
```diff
@@ -25,150 +25,164 @@
...
 
-#include <string.h> //memcpy (linux)
-#include <stdlib.h> //exit
+#include <stdlib.h>  //exit
+#include <string.h>  //memcpy (linux)
 
 template <typename atype>
-struct darray
-{
+struct darray {
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/darray.h]: Oh lord, this is still used? Wow....

File: ENIGMAsystem/SHELL/Universal_System/depth_draw.h [hunk 2]
```diff
@@ -28,35 +28,39 @@
...
-  typedef map<double,depth_layer>::reverse_iterator diter;
-}
-#endif
+struct depth_layer {
+  std::vector<tile> tiles;
+  event_iter* draw_events;
+  int tilelist;
+
+  //TODO: This should probably be moved into graphics_systems/GENERAL?
+  std::vector<std::vector<int> >
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/depth_draw.h]: This is why I don't trust everything to the formatter.

File: ENIGMAsystem/SHELL/Universal_System/dynamic_args.h [hunk 3]
```diff
@@ -25,23 +25,26 @@
...
-    varargs(variant);
-    varargs(varargs&);
-    ~varargs();
-    #endif
-  };
-}
+struct varargs {
+  int argc;
+  void* argv;
+#ifndef JUST_DEFINE_IT_RUN
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/dynamic_args.h]: And, while it's now more consistent with Google style, this looked better before, honestly. No point to formatting this file.

File: ENIGMAsystem/SHELL/Universal_System/fileio.cpp [hunk 4]
```diff
@@ -157,92 +158,91 @@ double file_text_read_real(int fileid) { // Reads a real value from the file and
...
-    if (ret[ret.length()-1] == '\n' or ret[ret.length()-1] == '\r')
-      break;
+    if (ret[ret.length() - 1] == '\n' or ret[ret.length() - 1] == '\r') break;
     buf[0] = 0;
   }
   size_t dp;
-  for (dp = ret.length()-1; dp != size_t(-1) and (ret[dp] == '\n' or ret[dp] == '\r'); dp--);
-  ret.erase(dp+1);
+  for (dp = ret.length() - 1; dp != size_t(-1) and (ret[dp] == '\n' or ret[dp] == '\r'); dp--)
+    ;
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/fileio.cpp]: ew

File: ENIGMAsystem/SHELL/Universal_System/fileio.cpp [hunk 5]
```diff
@@ -17,116 +17,118 @@
...
 
 #include "estring.h"
 
-namespace enigma_user
-{
+namespace enigma_user {
 
-int file_text_open_read(string fname) // Opens the file with the indicated name for reading. The function returns the id of the file that must be used in the other functions. You can open multiple files at the same time (32 max). Don't forget to close them once you are finished with them.
+int file_text_open_read(
+    string
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/fileio.cpp]: ew

File: ENIGMAsystem/SHELL/Universal_System/fileio.cpp [hunk 6]
```diff
@@ -17,116 +17,118 @@
...
+        fname)  // Opens the file with the indicated name for reading. The function returns the id of the file that must be used in the other functions. You can open multiple files at the same time (32 max). Don't forget to close them once you are finished with them.
 {
-  FILE *a = fopen(fname.c_str(),"rt"); //Read as text file
-  if (!a)      //Failure
-    return -1; //Behavior for fail is -1 return
-  enigma::files[enigma::file_highid] = enigma::openFile(a,""); //Store it in the lowest available ID, highid
+  FILE *a = fopen(fname.c_str(), "rt");                          //Read as text file
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/fileio.cpp]: This is actually kinda cool.

File: ENIGMAsystem/SHELL/Universal_System/gif_format.cpp [hunk 7]
```diff
@@ -19,29 +19,29 @@
 #include "gif_format.h"
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/gif_format.cpp]: Did you just format this file without changing it? All that does is gunk up the blame layer.

File: ENIGMAsystem/SHELL/Universal_System/Extensions/Alarms/alarmcode.cpp [hunk 8]
```diff
@@ -7,8 +7,6 @@
 #include "implement.h"
 #include "include.h"
 
-extern bool argument_relative;
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/Alarms/alarmcode.cpp]: This has broken the fps example by Mark Overmars:

```
C:/Users/Owner/AppData/Local/ENIGMA/Preprocessor_Environment_Editable/IDE_EDIT_objectfunctionality.h:219:5: error: reference to 'argument_relative' is ambiguous
     argument_relative = 0;
     ^
```

File: ENIGMAsystem/SHELL/Universal_System/GAME_GLOBALS.h [hunk 9]
```diff
@@ -29,27 +29,16 @@
 GM Global variables
 ******************/
 
-#ifndef __GAME_GLOBALS_H
-#define __GAME_GLOBALS_H
+#ifndef ENIGMA_GAME_GLOBALS_H
+#define ENIGMA_GAME_GLOBALS_H
 
-bool argument_relative=false;
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Universal_System/GAME_GLOBALS.h]: Moving this into enigma_user will break, and has broken, drag and drop:
https://github.com/enigma-dev/enigma-dev/pull/1128/files#r161388840

**Pull request closed without merging**